mempool/src/pool/mod.rs

```diff
@@ -1104,5 +1154,58 @@ impl<M: MemoryUsageEstimator> Mempool<M> {
     }
 }
 
+fn get_closest_value<V: Copy>(size_to_score: &BTreeMap<usize, V>, key: usize) -> Option<V> {
```

Comment:
Can we use linear interpolation instead of this "closest" value? The problem here is that the error is really unbounded, especially at the boundaries.

Now this is not that hard to do. Two things need to happen to make this work:

1. `key` has to be bounded such that it's within the min and max of that map.
2. The value can be calculated with a fixed formula that can be found here under the "Linear interpolation between two known points": https://en.wikipedia.org/wiki/Linear_interpolation

Just slap the formula and it'll be fine.

Please write tests for this as well that ensure that the values make sense. So getting the value at a known point should yield that point, and in-between two points should be the value between them.

Finally, the linear interpolation function can be thrown into utils... since it has to be used also in the API server.